packages/ts/react-form/test/useForm.spec.tsx

```diff
@@ -321,5 +321,36 @@ describe('@vaadin/hilla-react-form', () => {
       await user.click(loadFormBtn);
       await expect(findByTestId('contracts')).to.eventually.have.nested.property('selectedOptions.0.value', '202');
     });
+
+    it('runs validators on change event', async () => {
+      function ProjectForm() {
+        const { addValidator, field, model } = useForm(EntityModel);
+        const [count, setCount] = useState(0);
+
+        useEffect(() => {
+          addValidator({
+            message: 'ID expected',
+            validate: (entity) => {
+              setCount(count + 1);
+              return !entity.projectId;
+            },
+          });
+        }, []);
+
+        return (
+          <>
+            <input type="number" data-testid="project" {...field(model.projectId)} />
+            <div data-testid="count">{count.toString()}</div>
+          </>
+        );
+      }
+
+      const { findByTestId } = render(<ProjectForm />);
+      const projectInput = await findByTestId('project');
+      await user.type(projectInput, '123');
+      // Mimic change that happens without blur
+      fireEvent.change(projectInput);
+      await expect(findByTestId('count')).to.eventually.have.property('textContent', '1');
```

Comment:
As far as I remember, you can shrink it to just (maybe some Chai plugin is required).
```suggestion
      await expect(findByTestId('count')).to.eventually.have.text('1');
```